dbt/adapters/athena/impl.py

```diff
@@ -452,27 +447,34 @@ def _get_catalog_schemas(self, manifest: Manifest) -> AthenaSchemaSearchMap:
             info_schema_name_map.add(relation)
         return info_schema_name_map
 
-    def _get_data_catalog(self, catalog_name):
-        conn = self.connections.get_thread_connection()
-        client = conn.handle
-        with boto3_client_lock:
-            athena_client = client.session.client("athena", region_name=client.region_name, config=get_boto3_config())
+    def _get_data_catalog(self, database: str) -> Optional[DataCatalogTypeDef]:
+        if database:
+            conn = self.connections.get_thread_connection()
+            client = conn.handle
+            if database.lower() == "awsdatacatalog":
+                with boto3_client_lock:
+                    sts = client.session.client("sts", region_name=client.region_name, config=get_boto3_config())
+                catalog_id = sts.get_caller_identity()["Account"]
+                return {"Name": database, "Type": "GLUE", "Parameters": {"catalog-id": catalog_id}}
+            else:
+                with boto3_client_lock:
+                    athena = client.session.client("athena", region_name=client.region_name, config=get_boto3_config())
+                return athena.get_data_catalog(Name=database)["DataCatalog"]
 
-        response = athena_client.get_data_catalog(Name=catalog_name)
-        return response["DataCatalog"]
+    @staticmethod
+    def _get_catalog_id(catalog: Optional[DataCatalogTypeDef]) -> Optional[str]:
+        if catalog:
+            return catalog["Parameters"]["catalog-id"]
 
     def list_relations_without_caching(
         self,
         schema_relation: AthenaRelation,
     ) -> List[BaseRelation]:
-        catalog_id = None
-        if schema_relation.database is not None and schema_relation.database.lower() != "awsdatacatalog":
-            data_catalog = self._get_data_catalog(schema_relation.database.lower())
+        data_catalog = self._get_data_catalog(schema_relation.database)
+        catalog_id = self._get_catalog_id(data_catalog)
```

Comment:
If you are using `staticmethod` why still use the `self. _get_catalog_id ` ? I think it makes the code more confusing.